overlord/snapstate/handlers.go

```diff
@@ -452,10 +452,29 @@ var (
 	mountPollInterval = 1 * time.Second
 )
 
+func otherSnapsLike(st *state.State, instanceName string) (bool, error) {
+	snapName, _ := snap.SplitInstanceName(instanceName)
+
+	all, err := All(st)
```

Comment:
we don't need to unmarshal the SnapStates,  code like NumSnaps or SyncCookis would work,  maybe we should have  a rawSnaps internal helper and use it in those places, otoh the patter is just 4 lines